sphinx_gallery/py_source_parser.py

```diff
@@ -68,32 +69,55 @@ def split_code_and_text_blocks(source_file):
 
     Returns
     -------
+    file_conf : dict
+        File-specific settings given in comments as:
+        # sphinx_gallery_<name> = <value>
     blocks : list of (label, content)
         List where each element is a tuple with the label ('text' or 'code'),
         and content string of block.
     """
-    docstring, rest_of_content = get_docstring_and_rest(source_file)
-    blocks = [('text', docstring)]
+    docstring, rest_of_content, lineno = get_docstring_and_rest(source_file)
+    blocks = [('text', docstring, 1)]
+
+    prop_pat = re.compile(
+        r"^\s*#\s*sphinx_gallery_([A-Za-z0-9_]+)\s*=\s*(.+)\s*$",
+        re.MULTILINE)
+    file_conf = {}
+    for match in re.finditer(prop_pat, rest_of_content):
+        name = match.group(1)
+        value = match.group(2)
+        try:
+            value = ast.literal_eval(value)
+        except (SyntaxError, ValueError):
+            logger.warning(
+                'Sphinx-gallery option %s was passed invalid value %s',
+                name, value)
+        else:
+            file_conf[name] = value
```

Comment:
We can also extend the test that checks for the thumbnail number with the line number config